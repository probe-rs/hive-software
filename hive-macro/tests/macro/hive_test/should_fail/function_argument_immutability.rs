#![allow(unused)]

use hive_test::HiveTargetInfo;
use hive_test::TestChannelHandle;
use probe_rs_test::Session;

#[hive_macro::hive_test]
fn test_function(
    test_channel: &mut dyn TestChannelHandle,
    session: &mut Session,
    target_info: &mut String,
) {
}

fn main() {}
