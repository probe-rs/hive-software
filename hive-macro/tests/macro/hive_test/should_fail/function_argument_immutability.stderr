error: Function argument should not be mutable

         = help: This argument in a Hive testfunction is an immutable reference.
         = note: example: fn my_fancy_test(test_channel: &mut dyn TestChannelHandle, session: &mut Session, target_info: &HiveTargetInfo);

  --> tests/macro/hive_test/should_fail/function_argument_immutability.rs:11:18
   |
11 |     target_info: &mut String,
   |                  ^
